### PR TITLE
chore: rename upgrade timeout

### PR DESCRIPTION
**Pull request by charleenfei**
## Description

This is a tiny fix which is the first step to address updating the `SendPacket` API noted in [this issue](https://github.com/cosmos/ibc-go/issues/3551), as well as for finishing [this pr](https://github.com/cosmos/ibc-go/pull/3535)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

**Review by colin-axner — APPROVED** [review]

File: proto/ibc/core/channel/v1/channel.proto [hunk 1]
```diff
@@ -168,3 +168,13 @@ message Acknowledgement {
     string error  = 22;
   }
 }
+
+// Timeout defines an exeuction deadline structure for 04-channel msg handlers.
+// This includes packet lifecycle handlers as well as handshake and upgrade protocol handlers.
+// A valid Timeout contains either one or both of a timestamp and block height (sequence).
+message Timeout {
+  // block height after which the upgrade times out
```
Comment (crodriguezvega) [inline proto/ibc/core/channel/v1/channel.proto:176]:
```suggestion
  // block height after which the packet or upgrade times out
```

File: proto/ibc/core/channel/v1/channel.proto [hunk 2]
```diff
@@ -168,3 +168,13 @@ message Acknowledgement {
...
+  ibc.core.client.v1.Height height = 1 [(gogoproto.nullable) = false];
+  // block timestamp (in nanoseconds) after which the upgrade times out
```
Comment (crodriguezvega) [inline proto/ibc/core/channel/v1/channel.proto:178]:
```suggestion
  // block timestamp (in nanoseconds) after which the packet or upgrade times out
```

**Review by crodriguezvega — COMMENTED** [review]
Thanks, @charleenfei.

File: modules/core/04-channel/types/upgrade.go [hunk 3]
```diff
@@ -28,8 +28,8 @@ func NewUpgradeFields(ordering Order, connectionHops []string, version string) U
 }
 
 // NewUpgradeTimeout returns a new UpgradeTimeout instance.
-func NewUpgradeTimeout(height clienttypes.Height, timestamp uint64) UpgradeTimeout {
-	return UpgradeTimeout{
+func NewUpgradeTimeout(height clienttypes.Height, timestamp uint64) Timeout {
```
Comment (damiannolan) [inline modules/core/04-channel/types/upgrade.go:31]: Should be `NewTimeout` but I see this is removed in #3573 👍

**Review by damiannolan — APPROVED** [review]
There is a test helper in `testing/endpoint.go` for `ChanUpgradeInit` which takes both timeout args separately. 

I think we can update that to make use of the new `Timeout` type also. 

LGTM!

**Review by damiannolan — APPROVED** [review]
There is a test helper in `testing/endpoint.go` for `ChanUpgradeInit` which takes both timeout args separately. 

I think we can update that to make use of the new `Timeout` type also. 

LGTM!

File: modules/core/04-channel/keeper/upgrade.go [hunk 4]
```diff
@@ -86,7 +86,7 @@ func (k Keeper) WriteUpgradeTryChannel(
 }
 
 // constructProposedUpgrade returns the proposed upgrade from the provided arguments.
-func (k Keeper) constructProposedUpgrade(ctx sdk.Context, portID, channelID string, fields types.UpgradeFields, timeout types.UpgradeTimeout) (types.Upgrade, error) {
+func (k Keeper) constructProposedUpgrade(ctx sdk.Context, portID, channelID string, fields types.UpgradeFields, timeout types.Timeout) (types.Upgrade, error) {
```
Comment (DimitrisJim) [inline modules/core/04-channel/keeper/upgrade.go:89]: minor consistency nit:
```suggestion
func (k Keeper) constructProposedUpgrade(ctx sdk.Context, portID, channelID string, fields types.UpgradeFields, upgradeTimeout types.Timeout) (types.Upgrade, error) {
```

**Pull request merged**